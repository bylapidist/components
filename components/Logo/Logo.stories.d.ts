import type { StoryObj } from '@storybook/react';
import { BaseProps } from '../types';
import { LogoProps } from './index';
declare const _default: import("@storybook/types").ComponentAnnotations<import("@storybook/react/dist/types-0a347bb9").R, import("@storybook/types").Args>;
export default _default;
export declare const FadeAnimation: StoryObj<BaseProps & LogoProps>;
export declare const SlideAnimation: StoryObj<BaseProps & LogoProps>;
