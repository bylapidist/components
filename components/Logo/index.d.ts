import { BaseProps } from '../types';
export type LogoAnimationType = 'fade' | 'slide';
export interface LogoProps {
    readonly animation?: LogoAnimationType;
}
declare const Logo: {
    ({ className, as, testId, animation, ...restProps }: BaseProps & LogoProps): import("react/jsx-runtime").JSX.Element;
    displayName: string;
};
export default Logo;
