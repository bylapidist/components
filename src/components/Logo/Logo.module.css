@import '../global.module.scss';

@keyframes fadeIn {
    0% { opacity: 0 }
    50% { opacity: 0 }
    100% { opacity: 1 }
}

@keyframes thinking {
    0% { opacity: 1 }
    50% { opacity: 0.6 }
    100% { opacity: 1 }
}

@keyframes slideRight {
    0% { transform: translate3d(-100%,0,0) }
    80% { transform: translateZ(0) }
    90% { transform: translate3d(-5%,0,0) }
    to { transform: translateZ(0) }
}

@keyframes slideUpLeft {
    0% { transform: translate3d(100%,100%,0) }
    80% { transform: translateZ(0) }
    90% { transform: translate3d(5%,5%,0) }
    to { transform: translateZ(0) }
}

@keyframes slideDown {
    0% { transform: translate3d(0,-100%,0) }
    80% { transform: translateZ(0) }
    90% { transform: translate3d(0,-5%,0) }
    to { transform: translateZ(0) }
}

.Logo {
    width: var(--size__16);
    overflow: hidden;
    animation: fadeIn;
    animation-duration: 0.5s;
    animation-iteration-count: 1;
    animation-timing-function: ease-in;
}

.Logo--thinking {
    animation: thinking;
    animation-duration: 0.8s;
    animation-iteration-count: infinite;
    animation-timing-function: ease-in-out;
}

.Logo--bUp {
    fill: var(--color-tertiary__dark);
    animation-delay: 0.1s;
}

.Logo--bDown {
    fill: var(--color-primary__dark);
    animation-delay: 0.2s;
}

.Logo--d {
    fill: var(--color-secondary__dark);
    animation-delay: 0.3s;
}

.Logo--bUp-animated {
    animation: slideRight;
    animation-duration: 0.8s;
    animation-iteration-count: 1;
    animation-timing-function: linear;
}

.Logo--bDown-animated {
    animation: slideUpLeft;
    animation-duration: 0.8s;
    animation-iteration-count: 1;
    animation-timing-function: linear;

}

.Logo--d-animated {
    animation: slideDown;
    animation-duration: 0.6s;
    animation-iteration-count: 1;
    animation-timing-function: linear;
}
