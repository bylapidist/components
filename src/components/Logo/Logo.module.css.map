{"version":3,"sourceRoot":"","sources":["../global.module.scss","Logo.module.scss"],"names":[],"mappings":"AAEA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;EAaI;EACA;EACA;EACA;EACA;EACA;EACA;;;AAIJ;EAAO;;;AACP;EAAuB;;;AACvB;AAAA;EACmD;;;AACnD;EAAY;;;AACZ;EAAO;EAAgB;;;AACvB;EAAe;;;AACf;EAAgB;;;AAChB;AAAA;EACoB;;;AACpB;EAAQ;EAA2B;;;AACnC;EAAM;EAAgB;;;AACtB;EAAY;EAAuB;EAAqB;;;AACxD;EAAwB;;;ACpCxB;EACI;IAAK;;EACL;IAAM;;EACN;IAAO;;;AAGX;EACI;IAAK;;EACL;IAAM;;EACN;IAAO;;;AAGX;EACI;IAAK;;EACL;IAAM;;EACN;IAAM;;EACN;IAAK;;;AAGT;EACI;IAAK;;EACL;IAAM;;EACN;IAAM;;EACN;IAAK;;;AAGT;EACI;IAAK;;EACL;IAAM;;EACN;IAAM;;EACN;IAAK;;;AAGT;EACI;EACA;EACA;EACA;EACA;EACA;;AAEA;EACI;;AAGJ;EACI;;AAGJ;EACI;;AAMA;EACI;EACA;EACA;EACA;EACA;;AAGJ;EACI;EACA;EACA;EACA;EACA;;AAGJ;EACI;EACA;EACA;EACA;EACA","file":"Logo.module.css"}