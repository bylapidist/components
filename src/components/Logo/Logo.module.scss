@import '../global.module';

@keyframes fade-in {
    0% { opacity: 0 }
    50% { opacity: 0 }
    100% { opacity: 1 }
}

@keyframes thinking {
    0% { opacity: 1 }
    50% { opacity: 0.6 }
    100% { opacity: 1 }
}

@keyframes slide-right {
    0% { transform: translate3d(-100%,0,0) }
    80% { transform: translateZ(0) }
    90% { transform: translate3d(-5%,0,0) }
    100% { transform: translateZ(0) }
}

@keyframes slide-up-left {
    0% { transform: translate3d(100%,100%,0) }
    80% { transform: translateZ(0) }
    90% { transform: translate3d(5%,5%,0) }
    100% { transform: translateZ(0) }
}

@keyframes slide-down {
    0% { transform: translate3d(0,-100%,0) }
    80% { transform: translateZ(0) }
    90% { transform: translate3d(0,-5%,0) }
    100% { transform: translateZ(0) }
}

.this {
    width: var(--size-16);
    overflow: hidden;
    animation: fade-in;
    animation-duration: 0.5s;
    animation-iteration-count: 1;
    animation-timing-function: ease-in;

    .a {
        fill: var(--colour-success-dark);
    }

    .b {
        fill: var(--colour-primary-dark);
    }

    .c {
        fill: var(--colour-secondary-dark);
    }

    &.animation-slide {
        .a {
            animation: slide-right;
            animation-delay: 0.1s;
            animation-duration: 0.8s;
            animation-iteration-count: 1;
            animation-timing-function: linear;
        }

        .b {
            animation: slide-up-left;
            animation-delay: 0.2s;
            animation-duration: 0.8s;
            animation-iteration-count: 1;
            animation-timing-function: linear;
        }

        .c {
            animation: slide-down;
            animation-delay: 0.3s;
            animation-duration: 0.6s;
            animation-iteration-count: 1;
            animation-timing-function: linear;
        }
    }
}
