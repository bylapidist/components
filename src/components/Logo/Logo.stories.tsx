import React from 'react';
import type { Meta, StoryObj, StoryFn } from '@storybook/react';
import { BaseProps } from '../types';
import ThemeProvider from '../ThemeProvider';
import Logo, { LogoProps } from './index';

export default {
    title: 'Primitives/Logo',
    component: Logo,
    decorators: [(getStory) => <ThemeProvider>{getStory()}</ThemeProvider>]
} as Meta;

const defaultArgs: BaseProps & LogoProps = {
    as: 'div',
    testId: 'Logo',
    animation: 'fade'
};

const DefaultTemplate: StoryFn<BaseProps & LogoProps> = (args) => (
    <Logo {...args} />
);

export const FadeAnimation: StoryObj<BaseProps & LogoProps> =
    DefaultTemplate.bind({});
FadeAnimation.args = {
    ...defaultArgs
};

export const SlideAnimation: StoryObj<BaseProps & LogoProps> =
    DefaultTemplate.bind({});
SlideAnimation.args = {
    ...defaultArgs,
    animation: 'slide'
};
