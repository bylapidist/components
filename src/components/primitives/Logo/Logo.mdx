import { Meta, ArgsTable, Story, Source, Canvas } from '@storybook/blocks';

import * as LogoStories from './Logo.stories.tsx';

<Meta of={LogoStories} />

# Logo

> A primitive for rendering the Lapidist logo. The logo always 'appears' rather than render immediately.

## Example

<Source dark />

## Controlling animation

The `animation` prop controls the way the logo appears. There are two available gutters, defined as `fade` and `slide`.

### Fade

> Use the `fade` animation for logos that aren't the primary logo of the page.

<Canvas>
    <Story of={LogoStories.FadeAnimation} />
</Canvas>

### Slide

> Use the `slide` animation for the primary logo on the page (e.g. within the header).

<Canvas>
    <Story of={LogoStories.SlideAnimation} />
</Canvas>

## Props

<ArgsTable />
