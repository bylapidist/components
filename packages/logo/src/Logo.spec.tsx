import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Logo from './index';

test('it works', () => {
    const tree = renderer
        .create(
            <Logo>
                <p>Hello world!</p>
            </Logo>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
