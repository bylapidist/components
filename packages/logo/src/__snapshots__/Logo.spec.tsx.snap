// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
.c1 {
  fill: #72af5d;
}

.c2 {
  fill: #17a3a5;
  -webkit-animation-delay: 0.1s;
  animation-delay: 0.1s;
}

.c3 {
  fill: #ebc764;
  -webkit-animation-delay: 0.2s;
  animation-delay: 0.2s;
}

.c0 {
  position: relative;
  display: block;
  width: 32px;
  height: 32px;
}

<svg
  className="c0"
>
  <polygon
    className="c1"
    color="#72af5d"
    points="0,0 0,16 16,16"
  />
  <polygon
    className="c2"
    color="#17a3a5"
    points="0,16 0,32 16,32"
  />
  <polygon
    className="c3"
    color="#ebc764"
    points="16,0 16,32 32,16"
  />
</svg>
`;

exports[`it works animated 1`] = `
.c1 {
  fill: #72af5d;
  -webkit-animation-duration: 0.8s;
  animation-duration: 0.8s;
  -webkit-animation-name: slide-right;
  animation-name: slide-right;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
}

.c2 {
  fill: #17a3a5;
  -webkit-animation-duration: 0.8s;
  animation-duration: 0.8s;
  -webkit-animation-name: slide-up-left;
  animation-name: slide-up-left;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
  -webkit-animation-delay: 0.1s;
  animation-delay: 0.1s;
}

.c3 {
  fill: #ebc764;
  -webkit-animation-duration: 0.8s;
  animation-duration: 0.8s;
  -webkit-animation-name: slide-down;
  animation-name: slide-down;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
  -webkit-animation-duration: 0.4s;
  animation-duration: 0.4s;
  -webkit-animation-delay: 0.2s;
  animation-delay: 0.2s;
}

.c0 {
  position: relative;
  display: block;
  width: 32px;
  height: 32px;
}

<svg
  className="c0"
>
  <polygon
    className="c1"
    color="#72af5d"
    points="0,0 0,16 16,16"
  />
  <polygon
    className="c2"
    color="#17a3a5"
    points="0,16 0,32 16,32"
  />
  <polygon
    className="c3"
    color="#ebc764"
    points="16,0 16,32 32,16"
  />
</svg>
`;

exports[`it works thinking 1`] = `
.c1 {
  fill: #72af5d;
  -webkit-animation-duration: 0.8s;
  animation-duration: 0.8s;
  -webkit-animation-name: thinking;
  animation-name: thinking;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
}

.c2 {
  fill: #17a3a5;
  -webkit-animation-duration: 0.8s;
  animation-duration: 0.8s;
  -webkit-animation-name: thinking;
  animation-name: thinking;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0.1s;
  animation-delay: 0.1s;
}

.c3 {
  fill: #ebc764;
  -webkit-animation-duration: 0.8s;
  animation-duration: 0.8s;
  -webkit-animation-name: thinking;
  animation-name: thinking;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0.2s;
  animation-delay: 0.2s;
}

.c0 {
  position: relative;
  display: block;
  width: 32px;
  height: 32px;
}

<svg
  className="c0"
>
  <polygon
    className="c1"
    color="#72af5d"
    points="0,0 0,16 16,16"
  />
  <polygon
    className="c2"
    color="#17a3a5"
    points="0,16 0,32 16,32"
  />
  <polygon
    className="c3"
    color="#ebc764"
    points="16,0 16,32 32,16"
  />
</svg>
`;

exports[`it works with different colours 1`] = `
.c1 {
  fill: #fff;
}

.c2 {
  fill: #fff;
  -webkit-animation-delay: 0.1s;
  animation-delay: 0.1s;
}

.c3 {
  fill: #fff;
  -webkit-animation-delay: 0.2s;
  animation-delay: 0.2s;
}

.c0 {
  position: relative;
  display: block;
  width: 32px;
  height: 32px;
}

<svg
  className="c0"
>
  <polygon
    className="c1"
    color="#fff"
    points="0,0 0,16 16,16"
  />
  <polygon
    className="c2"
    color="#fff"
    points="0,16 0,32 16,32"
  />
  <polygon
    className="c3"
    color="#fff"
    points="16,0 16,32 32,16"
  />
</svg>
`;
